### ISSUE TITLE
Couldn't create edge source (1:1)

### ISSUE THREAD

**Issue by dravorle** [Ovyerus/prismaliser#45]
This is a weird one. If I define the relation on my "UserProfile"-Model the console gets spammed by warnings:
`couldn't create edge for source handle id: User-UserToUserProfile-UserProfile; edge id: eUserToUserProfile`

If you however switch it around and define the relation on the side of the user model it works. Any Idea whats wrong there?

NOT WORKING:
```prisma
model User {
  id       String @id @unique @default(uuid()) @db.Char(36)
  username String @unique @db.VarChar(255)

  UserProfile UserProfile?
}

model UserProfile {
  id    String  @id @unique @db.Char(36)
  about String? @db.LongText

  User User @relation(fields: [id], references: [id])
}
```

WORKING:
```prisma
model User {
  id       String @id @unique @default(uuid()) @db.Char(36)
  username String @unique @db.VarChar(255)

  UserProfile UserProfile @relation(fields: [id], references: [id])
}

model UserProfile {
  id    String  @id @unique @db.Char(36)
  about String? @db.LongText
  
  User  User?
}
```

**Comment by Ovyerus** [comment]
Intriguing, I'll do some investigating and see if I can figure it out.

**Comment by villesau** [comment]
I encountered the same problem. Here's an example:

https://prismaliser.app?code=bW9kZWwgTW9kZWwxIHsKICBpZCAgICAgICAgIFN0cmluZyAgIEBpZCBAZGVmYXVsdChjdWlkKCkpCiAgY3JlYXRlZEF0ICBEYXRlVGltZSBAZGVmYXVsdChub3coKSkKICBtb2RpZmllZEF0IERhdGVUaW1lIEBkZWZhdWx0KG5vdygpKQogIG1vZGVsMiAgICAgTW9kZWwyPwp9Cgptb2RlbCBNb2RlbDIgewogIGlkICAgICAgIFN0cmluZyBAaWQgQGRlZmF1bHQoY3VpZCgpKQogIG1vZGVsMUlkIFN0cmluZyBAdW5pcXVlCiAgbW9kZWwxICAgTW9kZWwxIEByZWxhdGlvbihmaWVsZHM6IFttb2RlbDFJZF0sIHJlZmVyZW5jZXM6IFtpZF0pCn0%3D

**Comment by EdwardChristian** [comment]
Thank you for this tool @Ovyerus !

If it helps, I am also getting this bug. I am pasting the link here, the bug happens between the tables having a relation on userid 

https://prismaliser.app?code=Ly8gVGhpcyBpcyB5b3VyIFByaXNtYSBzY2hlbWEgZmlsZSwKLy8gbGVhcm4gbW9yZSBhYm91dCBpdCBpbiB0aGUgZG9jczogaHR0cHM6Ly9wcmlzLmx5L2QvcHJpc21hLXNjaGVtYQoKZ2VuZXJhdG9yIGNsaWVudCB7CiAgcHJvdmlkZXIgPSAicHJpc21hLWNsaWVudC1qcyIKfQoKZGF0YXNvdXJjZSBkYiB7CiAgcHJvdmlkZXIgPSAicG9zdGdyZXNxbCIKICB1cmwgICAgICA9IGVudigiREFUQUJBU0VfVVJMIikKfQoKbW9kZWwgVXNlciB7CiAgaWQgICAgICAgIEludCAgICAgICBAaWQgQGRlZmF1bHQoYXV0b2luY3JlbWVudCgpKQogIHVzZXJuYW1lICBTdHJpbmcgICAgQHVuaXF1ZQogIHBhc3N3b3JkICBTdHJpbmcKICBjcmVhdGVkQXQgRGF0ZVRpbWUgIEBkZWZhdWx0KG5vdygpKQogIGFjY291bnQgICBBY2NvdW50W10KICBzZXNzaW9uICAgU2Vzc2lvbj8KfQoKbW9kZWwgU2Vzc2lvbiB7CiAgaWQgICAgICAgIEludCAgICAgIEBpZCBAZGVmYXVsdChhdXRvaW5jcmVtZW50KCkpCiAgVXNlciAgICAgIFVzZXIgICAgIEByZWxhdGlvbihmaWVsZHM6IFt1c2VySWRdLCByZWZlcmVuY2VzOiBbaWRdKQogIHVzZXJJZCAgICBJbnQgICAgICBAdW5pcXVlCiAga2V5ICAgICAgIFN0cmluZwogIG1heEFnZSAgICBTdHJpbmcKICBjcmVhdGVkQXQgRGF0ZVRpbWUgQGRlZmF1bHQobm93KCkpCiAgdXBkYXRlZEF0IERhdGVUaW1lCn0KCm1vZGVsIEFjY291bnQgewogIGlkICAgICAgICAgICAgSW50ICAgICAgICAgICBAaWQgQGRlZmF1bHQoYXV0b2luY3JlbWVudCgpKQogIFVzZXIgICAgICAgICAgVXNlciAgICAgICAgICBAcmVsYXRpb24oZmllbGRzOiBbdXNlcklkXSwgcmVmZXJlbmNlczogW2lkXSkKICB1c2VySWQgICAgICAgIEludAogIHR3aXR0ZXJVc2VyICAgVHdpdHRlclVzZXIgICBAcmVsYXRpb24oZmllbGRzOiBbdHdpdHRlclVzZXJJZF0sIHJlZmVyZW5jZXM6IFtpZF0pCiAgdHdpdHRlclVzZXJJZCBCaWdJbnQKICB0d2VldCAgICAgICAgIFR3ZWV0W10KICB0d2l0dGVyT0F1dGggIFR3aXR0ZXJPQXV0aD8KfQoKbW9kZWwgVHdpdHRlclVzZXIgewogIGlkICAgICAgICAgICAgICBCaWdJbnQgICAgQGlkIEB1bmlxdWUKICBuYW1lICAgICAgICAgICAgU3RyaW5nCiAgc2NyZWVuTmFtZSAgICAgIFN0cmluZwogIHByb2ZpbGVJbWFnZVVybCBTdHJpbmcKICBhY2NvdW50cyAgICAgICAgQWNjb3VudFtdCiAgdHdlZXRzICAgICAgICAgIFR3ZWV0W10KfQoKbW9kZWwgVHdlZXQgewogIGlkICAgICAgICAgICAgSW50ICAgICAgICAgQGlkIEBkZWZhdWx0KGF1dG9pbmNyZW1lbnQoKSkKICBhY2NvdW50ICAgICAgIEFjY291bnQgICAgIEByZWxhdGlvbihmaWVsZHM6IFthY2NvdW50SWRdLCByZWZlcmVuY2VzOiBbaWRdKQogIGFjY291bnRJZCAgICAgSW50CiAgdHdpdHRlclVzZXIgICBUd2l0dGVyVXNlciBAcmVsYXRpb24oZmllbGRzOiBbdHdpdHRlclVzZXJJZF0sIHJlZmVyZW5jZXM6IFtpZF0pCiAgdHdpdHRlclVzZXJJZCBCaWdJbnQKICBjcmVhdGVkQXQgICAgIERhdGVUaW1lICAgIEBkZWZhdWx0KG5vdygpKQogIHVwZGF0ZWRBdCAgICAgRGF0ZVRpbWU_CiAgZGF0ZVRpbWUgICAgICBEYXRlVGltZSAgICBAZGIuVGltZXN0YW1wdHooMykKICBjb250ZW50ICAgICAgIFN0cmluZyAgICAgIEBkYi5WYXJDaGFyKDI4MCkKICBzZW50ICAgICAgICAgIEJvb2xlYW4gICAgIEBkZWZhdWx0KGZhbHNlKQogIHVwbG9hZHMgICAgICAgVXBsb2FkW10KfQoKbW9kZWwgVXBsb2FkIHsKICBpZCAgICAgIEludCAgICAgQGlkIEBkZWZhdWx0KGF1dG9pbmNyZW1lbnQoKSkKICB0d2VldCAgIFR3ZWV0ICAgQHJlbGF0aW9uKGZpZWxkczogW3R3ZWV0SWRdLCByZWZlcmVuY2VzOiBbaWRdKQogIHR3ZWV0SWQgSW50CiAgbmFtZSAgICBTdHJpbmcKICB1cmwgICAgIFN0cmluZyAgQHVuaXF1ZQogIGFsdCAgICAgU3RyaW5nPwogIHR5cGUgICAgU3RyaW5nCn0KCm1vZGVsIFR3aXR0ZXJPQXV0aCB7CiAgYWNjb3VudCAgICAgIEFjY291bnQgICBAcmVsYXRpb24oZmllbGRzOiBbYWNjb3VudElkXSwgcmVmZXJlbmNlczogW2lkXSkKICBhY2NvdW50SWQgICAgSW50ICAgICAgIEBpZCBAdW5pcXVlCiAgYWNjZXNzVG9rZW4gIFN0cmluZwogIGFjY2Vzc1NlY3JldCBTdHJpbmcKICBjcmVhdGVkQXQgICAgRGF0ZVRpbWUgIEBkZWZhdWx0KG5vdygpKQogIHVwZGF0ZWRBdCAgICBEYXRlVGltZT8KfQoKbW9kZWwgUGFwZXIgewogIGlkICAgICAgICAgICAgSW50ICAgICAgQGlkIEBkZWZhdWx0KGF1dG9pbmNyZW1lbnQoKSkKICB0eXBlICAgICAgICAgIFN0cmluZwogIHRpdGxlICAgICAgICAgU3RyaW5nCiAgYXV0aG9ycyAgICAgICBTdHJpbmdbXQogIGZ1bGxBdXRob3JzICAgU3RyaW5nPwogIGRvaSAgICAgICAgICAgU3RyaW5nICAgQHVuaXF1ZQogIHVybCAgICAgICAgICAgU3RyaW5nCiAgcHJlcHJpbnQgICAgICBTdHJpbmc_CiAgc2hvcnRBYnN0cmFjdCBTdHJpbmcKICBmdWxsQWJzdHJhY3QgIFN0cmluZz8KICBtb250aFllYXIgICAgIFN0cmluZz8KICBwYWdlcyAgICAgICAgIFN0cmluZz8KICBjaXRhdGlvbnMgICAgIEludD8KICBkb3dubG9hZHMgICAgIEludD8KfQo%3D

**Comment by KeentGG** [comment]
also encountered this bug. a 1:1 between user and profile.

https://prismaliser.app?code=bW9kZWwgVXNlciB7CiAgaWQgICAgICAgIFN0cmluZyAgIEBpZCBAZGVmYXVsdCh1dWlkKCkpCiAgY3JlYXRlZEF0IERhdGVUaW1lIEBkZWZhdWx0KG5vdygpKQogIHVwZGF0ZWRBdCBEYXRlVGltZSBAdXBkYXRlZEF0CiAgcHJvZmlsZSAgIFByb2ZpbGU_CgogIEBAbWFwKCJ1c2VycyIpCn0KCm1vZGVsIFByb2ZpbGUgewogIGlkICAgICBTdHJpbmcgQGlkIEBkZWZhdWx0KHV1aWQoKSkKICB1c2VySWQgU3RyaW5nIEB1bmlxdWUKICB1c2VyICAgVXNlciAgIEByZWxhdGlvbihmaWVsZHM6IFt1c2VySWRdLCByZWZlcmVuY2VzOiBbaWRdLCBvbkRlbGV0ZTogQ2FzY2FkZSkKCiAgQEBtYXAoInByb2ZpbGVzIikKfQo%3D

**Comment by Ovyerus** [comment]
Super sorry for dropping the ball on this; didn't have much motivation for programming outside of work for a long time, but I'm finally back on this, hoping to get the fix done in the coming days/weeks.

**Comment by Ovyerus** [comment]
Finally fixed in 38546cb37471d36e1e61cccd251950c6787178fa!
